### PR TITLE
#1-4Task.  Alina.K

### PR DESCRIPTION
**Pull request by kalinovska-ya**
https://travis-ci.com/github/kalinovska-ya/js-assignments/builds/174089287

### CODE REVIEW

File: task/01-strings-tasks.js [hunk 1]
```diff
@@ -99,7 +99,14 @@ function getFirstChar(value) {
...
  */
 function removeLeadingAndTrailingWhitespaces(value) {
-    throw new Error('Not implemented');
+    while (value.startsWith(' ') || value.startsWith('\t')) {
+        value = value.substr(1);
+    }
+    while (value.endsWith(' ') || value.endsWith('\t')) {
+        value = value.slice(0, -1);
+    }
+
```
Comment (IlyaMokin) [inline task/01-strings-tasks.js:108]: you can use a single regExp here or String.trim function

File: task/01-strings-tasks.js [hunk 2]
```diff
@@ -114,7 +121,12 @@ function removeLeadingAndTrailingWhitespaces(value) {
  *   'cat', 3 => 'catcatcat'
  */
 function repeatString(value, count) {
-    throw new Error('Not implemented');
+    let newValue = '';
+    for (let i = 1; i <= count; i++) {
+        newValue += value;
+    }
+
```
Comment (IlyaMokin) [inline task/01-strings-tasks.js:128]: please check String.repeat function

File: task/03-date-tasks.js [hunk 3]
```diff
@@ -56,7 +56,9 @@ function parseDataFromIso8601(value) {
  *    Date(2015,1,1)    => false
  */
 function isLeapYear(date) {
-   throw new Error('Not implemented');
+   let year = date.getFullYear();
+   let feb29 = new Date (year, 1, 29)
+   return (feb29.getMonth () == 1);
```
Comment (IlyaMokin) [inline task/03-date-tasks.js:61]: non-standard solution. Looks cool

**Review by IlyaMokin — COMMENTED** [review]
Looks good, but please check notes.
I'll mark the PR as "Draft", please click "ready for review" when it will be finished.
Thank you!

File: task/03-date-tasks.js (hunk 3 above)
Comment (kalinovska-ya) [reply task/03-date-tasks.js:61]: Thanks!

File: task/01-strings-tasks.js (hunk 1 above)
Comment (kalinovska-ya) [reply task/01-strings-tasks.js]: Updated.

File: task/01-strings-tasks.js (hunk 2 above)
Comment (kalinovska-ya) [reply task/01-strings-tasks.js]: Updated.

**Comment by kalinovska-ya** [comment]
https://travis-ci.com/github/kalinovska-ya/js-assignments/builds/175224270

**Comment by IlyaMokin** [comment]
Closing as accepted, thanks!

**Pull request closed without merging**